api/src/main/java/com/datastrato/gravitino/rel/types/Decimal.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.rel.types;
+
+import static com.datastrato.gravitino.rel.types.Types.DecimalType.checkPrecisionScale;
+
+import com.google.common.base.Preconditions;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Objects;
+
+/** Used to represent a {@link Types.DecimalType} value in Gravitino. */
+public class Decimal {
+
+  /**
+   * Creates a decimal value with the given value, precision and scale.
+   *
+   * @param value The value of the decimal.
+   * @param precision The precision of the decimal.
+   * @param scale The scale of the decimal.
+   * @return A new {@link Decimal} instance.
+   */
+  public static Decimal of(BigDecimal value, int precision, int scale) {
+    return new Decimal(value, precision, scale);
+  }
+
+  /**
+   * Creates a decimal value with the given value.
+   *
+   * @param value The value of the decimal.
+   * @return A new {@link Decimal} instance.
+   */
+  public static Decimal of(BigDecimal value) {
+    return new Decimal(value);
+  }
+
+  /**
+   * Creates a decimal value with the given value.
+   *
+   * @param value The value of the decimal.
+   * @return A new {@link Decimal} instance.
+   */
+  public static Decimal of(String value) {
+    return of(new BigDecimal(value));
+  }
+
+  /**
+   * Creates a decimal value with the given value, precision and scale.
+   *
+   * @param value The value of the decimal.
+   * @param precision The precision of the decimal.
+   * @param scale The scale of the decimal.
+   * @return A new {@link Decimal} instance.
+   */
+  public static Decimal of(String value, int precision, int scale) {
+    return of(new BigDecimal(value), precision, scale);
+  }
+
+  /** @return The value of the decimal. */
+  public BigDecimal value() {
+    return value;
+  }
+
+  /** @return The precision of the decimal. */
+  public int precision() {
+    return precision;
+  }
+
+  /** @return The scale of the decimal. */
+  public int scale() {
+    return scale;
+  }
+
+  private final BigDecimal value;
+  private final int precision;
+  private final int scale;
```

Comment:
Can you please move the class fields to the top of the class.

Reply:
done